pkg/kp/consulutil/watch.go

```diff
@@ -64,6 +65,69 @@ func WatchPrefix(
 	}
 }
 
+type PairsResult struct {
+	Pairs api.KVPairs
+	Err   error
+}
+
+// Like WatchPrefix(), but has a different and better control flow. WatchPrefix() takes its
+// two output channels (outPairs and outErrors) as arguments, as well as a quit channel.
+// This means that the caller of WatchPrefix is responsible both for closing the output
```

Comment:
`WatchPrefix()` does close its output channel [[L15](https://github.com/square/p2/pull/628/files#diff-bd93e88e13ecea60af2f8f61eddfc30aL15), [L27](https://github.com/square/p2/pull/628/files#diff-bd93e88e13ecea60af2f8f61eddfc30aL27)] when signaled to stop watching.